vcd/config.go

```diff
@@ -228,6 +228,18 @@ func (cli *VCDClient) unLockParentVapp(d *schema.ResourceData) {
 	vcdMutexKV.kvUnlock(key)
 }
 
+func (cli *VCDClient) lockVappWithName(org, vdc, vappName string) func() {
```

Comment:
Yep. I have also thought about it (and it would probably be a good thing), but then we have to replace it in all other places that lock vApp as otherwise these locks will not match. That is the reason I resisted making this PR large.
I could take on it, but IMO it is safer for next version

Reply:
True, no worries then!